File: app/src/main/java/com/execube/genesis/model/Movie.java
package com.execube.genesis.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by Prateek Phoenix on 4/24/2016.
 */
public class Movie extends RealmObject implements Parcelable {

    @PrimaryKey
    @SerializedName("id")
    private int id;


    @SerializedName("original_title")
    private String mOriginalTitle;

    @SerializedName("title")
    private String mTitle;

    @SerializedName("poster_path")
    private String mPosterPath;

    @SerializedName("overview")
    private String mOverview;

    @SerializedName("vote_average")
    private float mVoteAverage;

    @SerializedName("backdrop_path")
    private String mBackdropPath;

    @SerializedName("release_date")
    private String mReleaseDate;



    private Movie(Parcel source) {

        id =source.readInt();
        mOriginalTitle=source.readString();
        mTitle=source.readString();
        mPosterPath=source.readString();
        mOverview=source.readString();
        mVoteAverage=source.readFloat();
        mBackdropPath=source.readString();
        mReleaseDate=source.readString();

    }

    public Movie() {

    }

    public String getReleaseDate() {
        return mReleaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        mReleaseDate = releaseDate;
    }

    public int getMovieId() {
        return id;
    }

    public void setMovieId(int id) {
        this.id = id;
    }

    public String getOriginalTitle() {
        return mOriginalTitle;
    }

    public void setOriginalTitle(String originalTitle) {
        mOriginalTitle = originalTitle;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public String getPosterPath() {
        return mPosterPath;
    }

    public void setPosterPath(String posterPath) {
        mPosterPath = posterPath;
    }

    public String getOverview() {
        return mOverview;
    }

    public void setOverview(String overview) {
        mOverview = overview;
    }

    public float getVoteAverage() {
        return mVoteAverage;
    }

    public void setVoteAverage(float voteAverage) {
        mVoteAverage = voteAverage;
    }

    public String getBackdropPath() {
        return mBackdropPath;
    }

    public void setBackdropPath(String backdropPath) {
        mBackdropPath = backdropPath;
    }

    @Override
    public int describeContents() {
        return 0;
    }



    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(mOriginalTitle);
        dest.writeString(mTitle);
        dest.writeString(mPosterPath);
        dest.writeString(mOverview);
        dest.writeFloat(mVoteAverage);
        dest.writeString(mBackdropPath);
        dest.writeString(mReleaseDate);
    }

    public static final Parcelable.Creator<Movie> CREATOR= new Parcelable.Creator<Movie>(){

        @Override
        public Movie createFromParcel(Parcel source) {
            return new Movie(source);
        }

        @Override
        public Movie[] newArray(int size) {
            // FIXME: 28/04/16 it was creating empty array before
            return new Movie[size];
        }
    };


}

File: app/src/main/java/com/execube/genesis/model/Movie.kt
package com.execube.genesis.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Movie(
  @SerializedName("overview")
  var overview: String = "",
  @SerializedName("original_language")
  var originalLanguage: String = "",
  @SerializedName("original_title")
  var originalTitle: String = "",
  @SerializedName("video")
  var video: Boolean = false,
  @SerializedName("title")
  var title: String = "",
  @SerializedName("poster_path")
  var posterPath: String = "",
  @SerializedName("backdrop_path")
  var backdropPath: String = "",
  @SerializedName("release_date")
  var releaseDate: String = "",
  @SerializedName("vote_average")
  var voteAverage: Double = 0.0,
  @SerializedName("popularity")
  var popularity: Double = 0.0,
  @SerializedName("id")
  var id: Int = 0,
  @SerializedName("adult")
  var adult: Boolean = false,
  @SerializedName("vote_count")
  var voteCount: Int = 0,
  var isFavourite: Boolean = false
) : Parcelable